### PR TITLE
Add user management functionality

### PR DESCRIPTION
**Pull request by sachithKay**

### CODE REVIEW

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/apis/ResourceAPI.jsx [hunk 1]
```diff
@@ -138,4 +137,22 @@ export default class ResourceAPI {
...
+            isAdmin: isAdmin
+        };
+        return this.getHTTPClient().post(`/users`, payload);
+    }
+
+    deleteUser(userId) {
+        return this.getHTTPClient().delete(`/users/${userId}`)
+    }
+
+    getUserById(userId) {
```
Comment (sdkottegoda) [inline monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/apis/ResourceAPI.jsx:155]: Don't we need to add the method to getUsers by pattern and role?

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/DataUtils.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
```
Comment (GDLMadushanka) [inline monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/DataUtils.jsx:19]: shall we add a comment to explain this util usage

Comment (sachithKay) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/DataUtils.jsx:19]: fixed

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/apis/ResourceAPI.jsx (hunk 1 above)
Comment (sachithKay) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/src/utils/apis/ResourceAPI.jsx:155]: resolving as the offline discussion.

**Pull request merged**